pkg/csi/service/wcp/controller_helper.go

```diff
@@ -440,3 +443,88 @@ func getOverlappingNodes(accessibleNodes []string, topologyRequirement *csi.Topo
 	}
 	return overlappingNodes, nil
 }
+
+// canFakeAttach checks if the volume is eligible to be fake attached and returns a bool value
+func canFakeAttach(ctx context.Context, volumeID string, volumeManager cnsvolume.Manager) bool {
+	log := logger.GetLogger(ctx)
+	//Check volume annotations
+	pvcAnn, err := commonco.ContainerOrchestratorUtility.GetVolumeAnnotations(ctx, volumeID)
+	if err != nil {
+		log.Errorf("failed to get volume annotations")
+		return false
+	}
+
+	if val, found := pvcAnn[common.AnnIgnoreInaccessiblePV]; found && val == "yes" {
+		log.Infof("Found %s annotation for volume %s set to yes", common.AnnIgnoreInaccessiblePV, volumeID)
+		//Check if volume is inaccessible
+		log.Debugf("Checking volume health on CNS volume: %s", volumeID)
+		vol, err := common.QueryVolumeByID(ctx, volumeManager, volumeID)
+		if err != nil {
+			log.Errorf("failed to query CNS volume")
+			return false
+		}
+
+		if vol.HealthStatus != string(pbmtypes.PbmHealthStatusForEntityUnknown) {
+			volHealthStatus, err := convertVolumeHealthStatus(vol.HealthStatus)
+			if err != nil {
+				log.Errorf("invalid health status: %s for volume: %s", vol.HealthStatus, vol.VolumeId.Id)
+				return false
+			}
+			log.Infof("CNS volume health is: %s", volHealthStatus)
+
+			annotations := make(map[string]string)
+			// If CNS volume health doesn't match with volume health annotation on pvc,
+			// mark volume health on pvc to be updated.
+			if val, found := pvcAnn[common.AnnVolumeHealth]; !found || val != volHealthStatus {
+				annotations[common.AnnVolumeHealth] = volHealthStatus
+			}
+			// If volume is inaccessible, it can be fake attached.
+			// Mark fake-attached annotation on pvc to be updated.
+			if volHealthStatus == common.VolHealthStatusInaccessible {
+				annotations[common.AnnFakeAttached] = "yes"
+			}
+			// Update annotations
+			if err := commonco.ContainerOrchestratorUtility.UpdateVolumeAnnotations(ctx, volumeID, annotations); err != nil {
+				// If updating pvc annotation fails, the volume should not be fake attached
+				return false
+			}
+
+			// If volume is inaccessible, it can be fake attached.
+			return volHealthStatus == common.VolHealthStatusInaccessible
+		}
+		// For all other cases, return false
+		return false
+	}
+	// Annotation is not found or not set to true, return false
+	log.Infof("Annotation %s not found or not set to true for volume %s", common.AnnIgnoreInaccessiblePV, volumeID)
+	return false
+}
+
+// convertVolumeHealthStatus convert the volume health status into accessible/inaccessible status
+func convertVolumeHealthStatus(volHealthStatus string) (string, error) {
```

Comment:
We already have it here - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/9ff9abf85c9efe584b005131f55614f0eba94301/pkg/syncer/volume_health.go#L138. Can we not reuse?

Reply:
We can't reuse as we can't import `syncer` package in `k8sorchestrator`. It's resulting in cyclic dependency.
I plan to move methods and variables common across driver and syncer to a common package and use that. But that will be put in a subsequent PR.
```
package sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common/commonco/k8sorchestrator
	imports sigs.k8s.io/vsphere-csi-driver/pkg/syncer
	imports sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common/commonco
	imports sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common/commonco/k8sorchestrator: import cycle not allowed
```

Reply:
>I plan to move methods and variables common across driver and syncer to a common package and use that. But that will be put in a subsequent PR.

Why not do it now?

Reply:
Done.